code/game/objects/structures/displaycase.dm

```diff
@@ -10,13 +10,15 @@
 	max_integrity = 200
 	integrity_failure = 0.25
 	var/obj/item/showpiece = null
+	var/obj/item/showpiece_type = null //This allows for showpieces that can only hold items if they're the same istype as this.

```

Comment:
make this a typecache instead imo

Reply:
Yeah, for most usage I was intending on I'll really only need a single type instead of a typecache, but if that changes in the future that'll be the go-to option.